ACM#9988: Update configuring the HCP on bare metal topic (#6152)
open-cluster-management/rhacm-docs: clusters/hosted_control_planes/bm_intro.adoc

- To provision hosted control planes on bare metal, you can use the Agent platform. The Agent platform uses the central infrastructure management service to add worker nodes to a hosted cluster. For an introduction to the central infrastructure management service, see xref:../cluster_lifecycle/cim_enable.adoc#enable-cim[Enabling the central infrastructure management service].

> - All bare metal hosts require a manual boot with a Discovery Image ISO that the central infrastructure management provides. You can start the hosts manually or through automation by using `Cluster-Baremetal-Operator`. After each host starts, it runs an Agent process to discover the host details and complete the installation. An `Agent` custom resource represents each host.

Reviewer: I would say define ISO then provide the acronym (unless you already did so earlier in this doc). 

But then I looked at some RHEL documentation, and they don't define what ISO is, they simply use the acronym, so maybe it's a widely known enough acronym.

Reviewer: So, I'll defer to what you want to do :)

- When you create a hosted cluster with the Agent platform, HyperShift installs the Agent Cluster API provider in the hosted control plane namespace.

> - A machine is created for every replica you scale the node pool up by. For every machine, the Cluster API provider finds and installs an Agent that is approved, is passing validations, is not currently in use, and meets the requirements that are specified in the node pool specification. You can monitor the installation of an Agent by checking its status and conditions.

Reviewer: Make it more user focused. And some tweaks for minimalism. What do you think of the following: 

```suggestion
- When you scale a replica by the node pool, HyperShift creates a machine. For every machine, the Cluster API provider finds and installs an Agent that is: approved, passing validations, currently in use, and meets the requirements that are specified in the node pool specification. You can monitor the installation of an Agent by checking its status and conditions.
```

Author: Changed a sentence a bit and removed some part ("meets the requirements" covers all conditions)
